server/models/pattern/resource/selector/workload.go

```diff
@@ -29,15 +32,17 @@ func (s *Selector) Workload(name string, version string, model string) (v1alpha1
 				comp, ok = en.(v1alpha1.ComponentDefinition)
 				if ok {
 					found = true
+				}
+				if comp.Model.Version == version { //prefer to use the correct version, if available
```

Comment:
It makes sense to include a debug level log of the decision the engine makes of which registrant to use.

Reply:
After a log message is considered, this LGTM

Reply:
Okay